test/data/gauge/valid.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+call_endpoint() {
+    nc -w 1 -u 0.0.0.0 8125
+}
+
+############################
+# Simple correct case
+
+echo "login:1|g"        | call_endpoint
+echo "login:3|g"        | call_endpoint
+echo "login:5|g"        | call_endpoint
+echo "logout:4|g"       | call_endpoint
+echo "logout:2|g"       | call_endpoint
+echo "logout:2|g"       | call_endpoint
+echo "login:+0.5|g"     | call_endpoint
+echo "login:-0.12|g"    | call_endpoint
+echo "logout:0.128|g"   | call_endpoint
+
+## Results:
+## login = 5
+## logout = 2
+############################
+
+
+############################
+# Add and decrement cases
+
+echo "success:0|g"      | call_endpoint
+echo "success:+5|g"     | call_endpoint
+echo "success:-12|g"    | call_endpoint
+echo "error:0|g"        | call_endpoint
+echo "error:+9|g"       | call_endpoint
+echo "error:-0|g"       | call_endpoint
+
+## Results:
+## success = -7 
+## error = 9
+############################
```

Comment:
To write some performance tests it would be necessary to write a short script in perl, python, ruby or pretty much anything else instead calling `nc` for every UDP packet - that will be slow and probably the agent won't be saturated. Just a side note.

Reply:
Do you know of any example of such test? Looking around it seems like python tests often work as wrappers around shared libs. I would just write the tests in JS under Node but I don't think thats what would be OK with you.